contracts/ERC20Decrypto.sol

```diff
@@ -210,11 +231,257 @@ contract ERC20Decrypto is
             "ERC20: newAddressFee could not be 0"
         );
         //set old owner
-        address oldOwner = _addressFee;
+        address oldOwner = addressFee;
+        //set new owner
+        addressFee = newAddressFee;
+        //emit event
+        emit AddressFeeChange(oldOwner, addressFee);
+    }
+
+    /**
+     * @dev Set split
+     *
+     * Emits a {SplitChange} event.
+     * Requirements:
+     * - the caller must have the `ADMIN_ROLE`.
+     */
+    function split() public virtual {
+        require(
+            hasRole(DEFAULT_ADMIN_ROLE, _msgSender()),
+            "ERC20: must have admin role to split"
+        );
+        _setSplit(splitMultiplier.mul(2), splitDivider);
+    }
+
+    /**
+     * @dev Set halve split
+     *
+     * Emits a {SplitChange} event.
+     * Requirements:
+     * - the caller must have the `ADMIN_ROLE`.
+     */
+    function reverseSplit() public virtual {
+        require(
+            hasRole(DEFAULT_ADMIN_ROLE, _msgSender()),
+            "ERC20: must have admin role to reverseSplit"
+        );
+        _setSplit(splitMultiplier, splitDivider.mul(2));
+    }
+
+    /**
+     * @dev See {IERC20-balanceOf}.
+     */
+    function balanceOf(address account)
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        return _formattedValue(_balances[account]);
+    }
+
+    /**
+     * @dev See {IERC20-allowance}.
+     */
+    function allowance(address owner, address spender)
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        return _formattedValue(_allowances[owner][spender]);
+    }
+
+    /**
+     * @dev Atomically increases the allowance granted to `spender` by the caller.
+     *
+     * This is an alternative to {approve} that can be used as a mitigation for
+     * problems described in {IERC20-approve}.
+     *
+     * Emits an {Approval} event indicating the updated allowance.
+     *
+     * Requirements:
+     *
+     * - `spender` cannot be the zero address.
+     */
+    function increaseAllowance(address spender, uint256 addedValue)
+        public
+        virtual
+        override
+        returns (bool)
+    {
+        // uint256 splitValue = _underlyingValue(addedValue);
+        uint256 formattedValue =
+            _formattedValue(_allowances[_msgSender()][spender]);
+        _approve(
```

Comment:
instead of modifying increase allowance, and decrease allowance its easier just to modify _approve

Reply:
_approve was modified and expects a formatted number